pennylane/pauli/pauli_arithmetic.py

```diff
@@ -414,10 +440,61 @@ def _pw_wires(w: Iterable) -> wires.Wires:
 
         return math.expand_matrix(reduced_mat, result_wire_order, wire_order=wire_order)
 
+    def _to_sparse_mat(self, wire_order):
+        """Compute the sparse matrix of the Pauli sentence by efficiently adding the Pauli words
+        that conform it. See pauli_sparse_matrices.md for the technical details."""
+        pauli_words = list(self)  # Ensure consistent ordering
+        n_wires = len(wire_order)
+        matrix_size = 2**n_wires
+        matrix = sparse.csr_matrix((matrix_size, matrix_size), dtype="complex128")
+        op_sparse_idx = _ps_to_sparse_index(pauli_words, wire_order)
+        _, unique_inds, unique_invs = np.unique(
+            op_sparse_idx, axis=0, return_index=True, return_inverse=True
+        )
+        pw_sparse_structures = unique_inds[unique_invs]
+
+        buffer_size = 2 ** max(0, 31 - n_wires - 5)  # Buffer of 1GB memory
```

Comment:
Maybe we could extract 2**26 to be a global variable:
```
SPARSE_MATRIX_BUFFER_SIZE = 2**26
```

then do:
```
buffer_size = max(1, SPARSE_MATRIX_BUFFER_SIZE / (2**n_wires) )
```

That way a test could locally set and reset the global variable to something smaller, like:

```
qml.pauli.pauli_arithemtic.SPARSE_MATRIX_BUFFER_SIZE = 2**6
# testing 
qml.pauli.pauli_arithemtic.SPARSE_MATRIX_BUFFER_SIZE = 2**26
```

Reply:
Because this the `buffer_size` here is more like `max_terms_in_buffer`.

Reply:
Oh yes! That would also allow the users to tune the buffer size to their needs. Since this would be oriented for more advanced users, I think it'd be better than adding an extra kwarg to `PauliSentence.to_mat()`

Reply:
Transferring discussion from Slack, I found that `2**26` complex entries is what worked best for my laptop and the Scipy Sum overload. Beyond a certain point, Scipy's CSR Sum will take quite some time to merge/sort indices. The implementation in `_sum_different_structure_pws` always benefit from a bigger buffer (in my experience & on my laptop). As a user/dev, I would prefer setting `SPARSE_MATRIX_BUFFER_SIZE` in terms of bytes, so `2**30 = 1 GB`, and have the function handle the conversion to `2**26` complex entries under the hood. If my assumption ( `_sum_different_structure_pws` always benefit from a bigger buffer) is correct, we could also fetch the total RAM and use, say, 10% as a default (e.g. 1.6 GB on my laptop). I could test the behaviour on heftier platforms like Perlmutter if we go down that route.

Reply:
Having the keyword argument here for `buffer_size` instead of having a global variable would make it easier to tweak and optimize for those interested in optimizing it.

Reply:
> Having the keyword argument here for `buffer_size`

I agree, if we call `_to_sparse_mat` directly then that makes the most sense.